test/integration/link-with-hash/test/index.test.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import webdriver from 'next-webdriver'
+import {
+  findPort,
+  launchApp,
+  killApp,
+  nextStart,
+  nextBuild,
+} from 'next-test-utils'
+
+jest.setTimeout(1000 * 60 * 5)
+let app
+let appPort
+const appDir = join(__dirname, '..')
+
+const runTests = () => {
+  it('should not have hydration mis-match for hash link', async () => {
+    const browser = await webdriver(appPort, '/')
+    const browserLogs = await browser.log('browser')
+    let found = false
+    browserLogs.forEach((log) => {
+      if (
+        log.message.includes(
+          'Warning: Prop `%s` did not match. Server: %s Client: %s'
```

Comment:
Overall it's fine to check for just `Warning: Prop` to ensure it doesn't break accidentally later on. Especially as there should be no hydration mismatches by default 👍 

```suggestion
          'Warning: Prop'
```